addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3490,7 +3497,10 @@ registry.BackgroundOptimize = ImageHandlerOption.extend({
      */
     async _applyOptions() {
         await this._super(...arguments);
-        this.$target.css('background-image', `url('${this._getImg().getAttribute('src')}')`);
+        const img = this._getImg();
+        if (this._isImgOptimizable(img)) {
+            this.$target.css('background-image', `url('${img.getAttribute('src')}')`);
+        }
```

Comment:
The fix is the right one but its implementation could be better. Here the call to `_super` will make that `_isImgOptimizable` check too so you end up doing it twice. Could you improve all that? There are multiple solutions. Either make the base function returns a boolean and here get the result of the `_super` and check it to know if you have to do something or not. Or, maybe better, you can do this:

Base class:
```javascript
async _applyOptions() {
    const img = ...;;
    if (img cannot be optimized) {
         return;
    }
    this._forceOptions(img); // And this method you can overload in the BackgroundOptimize method
}
```
I am fine with everything, just don't duplicate the call to know if the image is optimizable.